### PR TITLE
Make newly added points editable

### LINKED ISSUE #2234: Newly added point in `Add` mode is selected but not editable
## 🐛 Bug

Hi,
Once a point has been drawn I would like to resize it immediately. Currently if I move the point size slider right after adding a new point, point size remains unchanged. I need to use the selection too, and select it and afterwards I can edit its size. So apparently right after creating a new point, the point is not selected, although it is highlighted as if it was selected. Would it be possible to make the point directly size-editable, without having first to select it?
Thanks
Benoit

## To Reproduce

1. Add a point in the graphical user interface.
2. Slide the point size slider --> the point does not get resized
3. Select the newly added point.
4. Slide the point size slider --> the point gets resized

## Expected behavior

The point should get resized at step 2 (in "To reproduce")

## Environment

 - Please copy and paste the information at napari info option in help menubar here:

napari: 0.4.3
Platform: macOS-11.2-x86_64-i386-64bit
Python: 3.9.1 | packaged by conda-forge | (default, Jan 26 2021, 01:32:59) [Clang 11.0.1 ]
Qt: 5.9.7
PyQt5: 5.9.2
NumPy: 1.19.5
SciPy: 1.6.0
Dask: 2021.01.1
VisPy: 0.6.6

OpenGL:
- GL version: 2.1 INTEL-16.1.10
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen #1: resolution 2560x1440, scale 2.0
- screen #2: resolution 1440x900, scale 2.0

Plugins:
- napari_plugin_engine: 0.1.9
- svg: 0.1.4

### PR DESCRIPTION
**Pull request by jamesyan-git**
# Description
As per Issue #2234 , newly added points are now selected and able to have their size, face color and edge color changed without re-selecting the point.

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References
Closes #2234 

# How has this been tested?

- [x] the test suite for my feature covers cases regarding changing the following attributes on a newly placed point: `size, face_color, edge_color`
- [x] all tests pass with my change

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality (tests are parameterized)
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

**Comment by jamesyan-git** [comment]
Hi @DragaDoncila &/or @kevinyamauchi
I believe this is an appropriate change, accompanied by appropriate testing and documentation. Please let me know if this is not the case, and if anything further needs to be done. Thanks 😄

File: napari/layers/points/_tests/test_points.py
Comment (DragaDoncila) [inline napari/layers/points/_tests/test_points.py]:
```suggestion
    """tests the newly placed points may be edited without re-selecting"""
```

File: napari/layers/points/_tests/test_points.py
Comment (DragaDoncila) [inline napari/layers/points/_tests/test_points.py]:
```suggestion
    layer = Points()
```
'Points' will be its default name so we don't need this here.

File: napari/layers/points/points.py
Comment (DragaDoncila) [inline napari/layers/points/points.py:626]: @kevinyamauchi I don't work much with points properties so not sure if this change makes sense or whether for properties we actually do still want to make it not editable in Add mode?

File: napari/layers/points/_tests/test_points.py
Comment (DragaDoncila) [inline napari/layers/points/_tests/test_points.py:2429]: Would be cool if we can add the same test for `edge_width` just for the sake of completeness!

**Review by DragaDoncila — CHANGES_REQUESTED** [review]
This looks great @jamesyan-git! Made some small change requests and pinged Kevin to have a look at the change for updating properties.

File: napari/layers/points/_tests/test_points.py
Comment (kevinyamauchi) [reply napari/layers/points/_tests/test_points.py:2429]: I agree!

File: napari/layers/points/points.py
Comment (kevinyamauchi) [reply napari/layers/points/points.py:626]: I think it makes sense to update the properties in `Add` mode if we are allowing that for the size, colors, etc., as it makes it consistent. Also, users may be coloring the faces/edges by properties so they will need to update the properties/features to update the color.

**Review by kevinyamauchi — COMMENTED** [review]
Thanks @jamesyan-git ! I think this is good to go once you address @DragaDoncila 's comments.

**Review by DragaDoncila — APPROVED** [review]
Congrats on your first PR @jamesyan-git! I'll leave it to @kevinyamauchi to do the merge honors 🤣

**Review by kevinyamauchi — APPROVED** [review]
LGTM! Thanks, @jamesyan-git !

**Pull request merged**
